Calculator/src/main/java/com/microsoft/device/display/samples/calculator/models/CalculatorModel.kt

```diff
@@ -0,0 +1,170 @@
+import androidx.compose.runtime.setValue
+import com.microsoft.device.display.samples.calculator.ui.pages.historyModel
+import kotlin.math.roundToInt
+
+class CalculatorModel() {
+    var x: String by mutableStateOf("0")
+    var y: String by mutableStateOf("0")
+    // if we are editing x or y
+    var isOnX: Boolean by (mutableStateOf(true))
+    // if we are overwriting the current variable by typing, or adding to it by typing
+    var overWrite: Boolean by (mutableStateOf(true))
+    var degrees: Boolean by (mutableStateOf(true))
+    var currentEquation: Equation by mutableStateOf(Equation.NONE)
+
+    private fun assertEquation() {
+        val prevX = this.x
+        val prevY = this.y
+        when (currentEquation) {
+            Equation.ADD -> add()
+            Equation.SUB -> sub()
+            Equation.MUL -> mul()
+            Equation.DIV -> div()
+            Equation.POW -> power()
+            Equation.MOD -> modulus()
+            else -> {}
+        }
+        decimalReformat()
+        if (currentEquation != Equation.NONE) historyModel.addToRecords(Record(prevX, prevY, currentEquation, this.x))
+        resetToX()
+        overWrite = true
+    }
+
+    // we pass this in instead of storing it as we only want to execute an action once per click
+    fun assertAction(ac: Action) {
+        when (ac) {
+            Action.DEL -> subValues()
+            Action.EQL -> assertEquation()
+            Action.CLR -> clearValues()
+            Action.INV -> inverse()
+            Action.SIN -> sin()
+            Action.COS -> cos()
+            Action.TAN -> tan()
+            Action.LOG -> log()
+            Action.LN -> ln()
+            Action.E -> e()
+            Action.PI -> pi()
+            Action.SQRT -> squareRoot()
+            Action.SQRD -> squared()
+            Action.DEG -> toggleDegree()
+            Action.PRCT -> percent()
+            Action.SIGN -> toggleNegative()
+            else -> {}
+        }
+        decimalReformat()
+    }
+
+    private fun toggleDegree() { degrees = !degrees }
+    private fun subValues() { if (isOnX) subX() else subY() }
+    private fun toggleNegative() { if (isOnX) { x = (x.toFloat() * -1.0f).toString() } else { y = (y.toFloat() * -1.0f).toString() } }
+
+    // remove hanging 0's and convert float to int when possible
+    private fun decimalReformat() {
+        val floatX = this.x.toFloat()
+        val floatY = this.y.toFloat()
+
+        if (floatX == 0.0f) { this.x = "0" } else if ((floatX - floatX.roundToInt()) == 0.0f) { this.x = (floatX.roundToInt()).toString() }
+
+        if (floatY == 0.0f) { this.y = "0" } else if ((floatY - floatY.roundToInt()) == 0.0f) { this.y = (floatY.roundToInt()).toString() }
+    }
+
+    private fun clearValues() {
+        // if the user clears when x is 0, we will do an all clear
+        if (x == "0" && isOnX) { historyModel.clearRecords() }
+        resetValues()
+    }
+    private fun resetToX() {
```

Comment:
style/readability - it's good to have spaces in between functions (if you ctrl + alt + L in all your files, it'll fix spacing issues for you)